content/blog/2020-01-vroom-1-1-0/index.Rmarkdown

```diff
@@ -0,0 +1,131 @@
+---
+title: vroom 1.1.0
+author: Jim Hester
+date: '2020-01-14'
+slug: vroom-1-0-0
+description: vroom 1.1.0 is now on CRAN!
+categories:
+  - package
+tags: [package, r-lib]
+photo:
+  url: https://www.pexels.com/photo/12801/
+  author: Chris Peeters
+---
+
+```{r include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+I'm excited to announce that [vroom 1.1.0](http://vroom.r-lib.org) is now
+available on CRAN!
+
+vroom reads rectangular data, such as comma separated (csv), tab separated (tsv) or fixed width files (fwf) into R.
+It performs similar roles to functions like [`readr::read_csv()`](http://readr.r-lib.org), [`data.table::fread()`](http://r-datatable.com) or `read.csv()`.
+But for many datasets `vroom::vroom()` can read them much, much faster (hence the name).
+Get the latest version with:
+
+```{r, eval = FALSE}
+install.packages("vroom")
+```
+
+And attach the package by running
+
+```{r}
+library(vroom)
+```
+
+Improvements in this release include a hex logo, support for big integer data, improved delimiter guessing, including delimiters in specifications, and streamlined reading from standard input.
+
+See the [change log](https://vroom.r-lib.org/news/index.html) for a full list of changes and bug fixes in this version.
+
+## Hex logo
+
+Thanks to [Allison Horst](https://github.com/allisonhorst/stats-illustrations) we now have an awesome hex logo for vroom!
+
+![](logo.png)
+
+## Big integer support
+
+R's standard integers are stored in 32 bits of binary data, which means that the largest value they can store is 2,147,483,647 (`2^31 - 1`).
+R implicitly converts integers for most operations with doubles to 64 bit floating point values, which is why you may not have noticed this limitation before.
+
+```{r}
+options(digits = 22)
+x <- 2147483647L
+str(x)
+str(x + 1L)
+str(x + 1)
+x + 1
+```
+
+However even 64 bit floating point values can only store consecutive integers up to 9,007,199,254,740,992 (`2^53`) without losing precision.
+You can observe this because if you try adding 1 to this number you will get the same number.
+
+```{r}
+y <- 9007199254740992
+z <- y + 1
+z
+y == z
+```
+
+To store consecutive integers bigger than this you need to use a 64 bit integer type.
+R does not have native support for 64 bit integers, however the [bit64](https://cran.r-project.org/package=bit64) package provides support for them.
+Because these integers are so large they rarely occur in real world data, however they can often be obtained from generated data, such as database identifiers.
```

Comment:
```suggestion
Because these integers are so large, they rarely occur in real world data, however they can often be obtained from generated data, such as database identifiers.
```